### ISSUE TITLE
Remote host debugging not working

### ISSUE THREAD

**Issue by AndrewNatoli** [felixfbecker/vscode-php-debug#11]
Hey Felix,

After getting to test with a project that isn't the included testproject I noticed that `localSourceRoot` was not being resolved properly when given an absolute path. 

`serverSourceRoot`: /var/www/
`localSourceRoot`:  /Developer/project/

See the screenshot below where the `cwd` of the adapter source directory was being combined with my `serverSourceRoot`. I imagine the actual project cwd is used when the adapter isn't being debugged?

![screen shot 2016-01-11 at 6 36 56 pm](https://cloud.githubusercontent.com/assets/4162073/12251879/4d6558dc-b8a0-11e5-85ba-a5723a3d1f1d.png)

This wasn't an issue when working with the testproject because it was relative to the adapter's source directory.

**Comment by felixfbecker** [comment]
I don't get why. `path.resolve(process.cwd(), '/Developer/project')` should always return `/Developer/project`, no matter what the cwd is.... And I always though btw cwd is always the VS Code project root

**Comment by felixfbecker** [comment]
@AndrewNatoli Could you please check if this issue still exists with latest master? If no, I would like to release it.

**Comment by AndrewNatoli** [comment]
Hey Felix, still seems to be a no-go when using mappings with the latest master.

- Local without mapping :heavy_check_mark: 

- Local with mapping :no_entry_sign: 
*breakpoint_set sent a lowercase path and four slashes*

- Remote with mapping :no_entry_sign: 
*breakpoint_set sent with a lowercase path and four slashes*

Currently looking into what's happening....

**Comment by felixfbecker** [comment]
@AndrewNatoli What settings are you using? I would like to sort this out now.
There is now a new boolean setting called `log` for launch.json that when enabled, logs all communication between VS Code and the adapter to the console even when not running in server mode. Can you please take the output you get there and post it here?

**Comment by AndrewNatoli** [comment]
Linux --> OSX testing on PHP 5.3 & XDebug 2.1

Using https://github.com/felixfbecker/vscode-php-debug/commit/01e26737b934f8b137c773c3a3c4079369a3da17

## config values
```
"serverSourceRoot": "/var/www/testproject/",
"localSourceRoot": "/Developer/lserv4/testproject/"
```

## Console output 
```
-> launchResponse
Response {
  seq: 0,
  type: 'response',
  request_seq: 2,
  command: 'launch',
  success: true }

-> initializedEvent
InitializedEvent { seq: 0, type: 'event', event: 'initialized' }

-> setBreakpointsRequest
{ type: 'request',
  seq: 3,
  command: 'setBreakpoints',
  arguments: 
   { source: { path: '/Developer/lserv4/testproject/index.php' },
     lines: [ 5 ] } }

-> setExceptionBreakpointsRequest
{ type: 'request',
  seq: 4,
  command: 'setExceptionBreakpoints',
  arguments: { filters: [ 'uncaught' ] } }

-> setExceptionBreakpointsResponse
Response {
  seq: 0,
  type: 'response',
  request_seq: 4,
  command: 'setExceptionBreakpoints',
  success: true }

-> setBreakpointsResponse
Response {
  seq: 0,
  type: 'response',
  request_seq: 3,
  command: 'setBreakpoints',
  success: true,
  body: { breakpoints: [ Breakpoint { verified: true, line: 5 } ] } }

-> threadEvent
ThreadEvent {
  seq: 0,
  type: 'event',
  event: 'thread',
  body: { reason: 'exited', threadId: 2 } }
```

## XDebug log
```
-> <init xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" fileuri="file:///var/www/testproject/index.php" language="PHP" protocol_version="1.0" appid="1116"><engine version="2.1.0"><![CDATA[Xdebug]]></engine><author><![CDATA[Derick Rethans]]></author><url><![CDATA[http://xdebug.org]]></url><copyright><![CDATA[Copyright (c) 2002-2010 by Derick Rethans]]></copyright></init>

<- feature_set -i 0 -n max_depth -v 1
-> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="feature_set" transaction_id="0" feature="max_depth" success="1"></response>

<- feature_set -i 1 -n max_children -v 9999
-> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="feature_set" transaction_id="1" feature="max_children" success="1"></response>

<- breakpoint_list -i 2
-> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="breakpoint_list" transaction_id="2"></response>

<- breakpoint_set -i 3 -t exception -x *
-> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="breakpoint_set" transaction_id="3" id="11160001"></response>

<- breakpoint_set -i 4 -t line -f file:///Developer/lserv4/testproject/index.php -n 5
-> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="breakpoint_set" transaction_id="4" id="11160002"></response>

<- run -i 5
-> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="run" transaction_id="5" status="stopping" reason="ok"></response>

<- stop -i 6
-> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="stop" transaction_id="6" status="stopped" reason="ok"></response>
```

TL;DR recap: The local file path is being set as the breakpoint.

Not exactly sure and don't have time at the moment to review everything that's changed since my PR, but my PR is a good "control" experiment as it worked in all of my test cases.

**Comment by felixfbecker** [comment]
Thanks, I'm trying to set up a VM right now so I hopefully can reproduce the issue.
But from your logs it definitely seems like there are no lowercase file paths coming from VS Code.

**Comment by AndrewNatoli** [comment]
Okay great.

The test I ran with a different project directory had them

```
<- breakpoint_set -i 2 -t line -f file:///developer/lserv4/****/message_browse.php -n 8
-> <response xmlns="urn:debugger_protocol_v1" xmlns:xdebug="http://xdebug.org/dbgp/xdebug" command="breakpoint_set" transaction_id="2" id="33250001"></response>
```

Not sure how changing the project directory made a difference but regardless it's just the wrong path being sent altogether.

**Comment by felixfbecker** [comment]
@AndrewNatoli I got everything set up with a Linux VM and was able to reproduce your issue. I made changes and got it all working now. Can you please check? However, I could not reproduce the issue with lowercased file names.

**Comment by AndrewNatoli** [comment]
Remote debugging is working for me on my usual Linux --> OSX setup! Awesome! 

I wouldn't worry about the lowercase paths for now.... as odd as they were, I couldn't reproduce them while working with my original changes where the right server path was sent. 

Thanks!

**Comment by felixfbecker** [comment]
Awesome! Then I will release it now. You should close [Micrsoft/vscode#1916](https://github.com/Microsoft/vscode/issues/1916) then
